### PR TITLE
turtlevcs: init at 0.6

### LINKED ISSUE #252317: Package request: turtle (gnome)
**Project description**

Turtle is a graphical interface for version control intended to run on gnome and nautilus.

Currently these dialogs are implemented:

- commit, add, stage, unstage, revert, resolve
- sync (pull+push), pull, push
- checkout, create branch
- init, clone
- remotes, submodules
- log, diff
- settings and about

**Metadata**

* homepage URL: https://gitlab.gnome.org/philippun1/turtle
* source URL: https://gitlab.gnome.org/philippun1/turtle
* license: gpl3+
* platforms: linux

### PR DESCRIPTION
**Pull request by BurNiinTRee**
Closes #252317

Adds a [turtle](https://gitlab.gnome.org/philippun1/turtle) package.
The nautilus plugin requires `gnome.nautilus-python` to be installed separately.

The package is a bit weird, since `turtle_cli` subcommands are implemented through subprocesses
of either `$PYTHON` or `sys.executable`. This doesn't work well with `buildPythonApplication`s hooks
for setting up python dependencies.

I'm not very familiar with neither python packaging in general, nor nixpkgs' infrastructure for it,
so there might be better ways of achieving this.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
The package needs a `meta` attribute.

File: pkgs/by-name/tu/turtle/package.nix
Comment (benneti) [inline pkgs/by-name/tu/turtle/package.nix:15]: I think it is more common to either drop this last , or do a `, ...`

**Review by benneti — CHANGES_REQUESTED** [review]
Is there a reason you choose the by name category? A similar program (gitkraken) is in `pkgs/applications/version-management/` so this might be better suited there as well?
Additionally,  I think you should also put a call to the program in `pkgs/top-level/all-packages.nix`.
After this it would also be much easier for me to test the PR

File: pkgs/by-name/tu/turtle/package.nix
Comment (benneti) [inline pkgs/by-name/tu/turtle/package.nix:80]: did you try instead of using the approach of splitting the package in two, setting `$out` as suffix or prefix to the for example `--prefix PYTHONPATH : $out/${remy-python.sitePackages}`

**Comment by FliegendeWurst** [comment]
> Is there a reason you choose the by name category?

It is recommended in the [documentation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package) (since #237439 was merged).

**Comment by benneti** [comment]
oops, thanks for the heads up!

**Comment by BurNiinTRee** [comment]
I removed the trailing comma and set `$PYTHONPATH` in the wrapper to get rid of the extra derivation.
Thanks for the suggetsion

**Review by benneti — APPROVED** [review]
Great, thank you I think it looks much nicer now!
I think before someone merges this you should sqush to two commits, one adding you as the maintainer and one for the init of the package.

**Comment by BurNiinTRee** [comment]
Done

**Comment by bryango** [comment]
Wow, this is nice! Although it has been sitting for a while... Let me see if ofborg complains:
@ofborg eval

**Review by bryango — APPROVED** [review]

File: pkgs/by-name/tu/turtle/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/tu/turtle/package.nix:60]: We cannot use $PYTHONPATH like this as it draws in build time dependencies

File: pkgs/by-name/tu/turtle/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/tu/turtle/package.nix:23]: Please use the new pyproject = true setting instead of format

File: pkgs/by-name/tu/turtle/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/tu/turtle/package.nix:16]: A new line here and there similar to other packages would help readability a lot.

File: pkgs/by-name/tu/turtle/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/tu/turtle/package.nix:70]:
```suggestion
    mainProgram = "turtle";
    maintainers = with maintainers; [ burniintree ];
```

**Pull request closed without merging**
